### PR TITLE
refactor: Break up superset/views/core.py

### PR DESCRIPTION
**Pull request by willbarrett**
### SUMMARY

The file `superset/views/core.py` is massive, to the point that I get lost in there all the time when I'm working on something. There's no need for us to stuff so much functionality into a single Python module. This PR tackles some low-hanging fruit for breaking up this file, moving out a few pieces that are logically separate from the elephant in the room: the `Superset` class.

I'm hoping that in a near-future PR I'll be able to start breaking up some of the larger functions in the `Superset` catch-all, but I wanted to get the community's reaction to these changes first.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TEST PLAN

CI! Boot up locally and make sure everything still works!

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### Reviewers
@john-bodley @mistercrunch @craig-rueda

### CODE REVIEW

**Review by craig-rueda — APPROVED** [review]
Looks like some checks are failing, but otherwise LGTM

File: .pylintrc [hunk 1]
```diff
@@ -48,7 +48,7 @@ unsafe-load-any-extension=no
 # A comma-separated list of package or module names from where C extensions may
 # be loaded. Extensions are loading into the active Python interpreter and may
 # run arbitrary code
-extension-pkg-whitelist=
+extension-pkg-whitelist=pyarrow
```
Comment (dpgaspar) [inline .pylintrc:51]: nice!

**Comment by willbarrett** [comment]
Woohoo! Green build! What do you think @john-bodley @craig-rueda - good to go?

**Comment by john-bodley** [comment]
@willbarrett thanks for the change.

Unrelated to this change (but relevant to the `superset.views` module) I think we should probably strive to use `base.py` to represent base classes which get extended and `core.py` (or similar) to represent core logic (including functions). I think at the moment there isn't a clear divide between the two.

**Comment by willbarrett** [comment]
@john-bodley agreed on centralizing base classes. As for `core.py`, I think `core.py` should turn into `superset.py` and only hold the `Superset` class. Ideally the majority of the functionality of that class would be extracted to Command or DAO objects. More on this in PR form in the near future I hope.

**Pull request merged**